### DISCUSSION TITLE
Canvas LMS integration with Slate CRM

### DISCUSSION THREAD

**Discussion by ShiroW0lf** [ucfopen/canvasapi#607] [Q&A] [answered]
Hi, I am from the University of Texas Permian Basin. We are currently trying to integrate the Canvas LMS with our Slate crm using the Canvas API. Our goal is to bring the to-do list from Canvas into our student portal on slate. Has anyone done something similar to this before. We would like to know the best approach to move forward with this project. Plus: any other tips and suggestions will be much appreciated. Thank You.

**Comment by bennettscience** [comment, marked as answer]
As far as getting the to do items, those are in the `Canvas` instance because [they're tied to the current user making the request](https://canvasapi.readthedocs.io/en/stable/canvas-ref.html?highlight=todo#canvasapi.Canvas.get_todo_items):

```python
from canvasapi import Canvas

canvas = Canvas(URL, KEY)
todos = canvas.get_todo_items()
# do more stuff
```

I don't see a way in the documents to get the todo list for all users. You may need to [rely on masquerading](https://canvas.instructure.com/doc/api/file.masquerading.html) in your script if you're trying to get specific lists for logged in users.

**Reply by ShiroW0lf** [reply]
Thank You !
